integrations/launchdarkly/client.py

```diff
@@ -0,0 +1,218 @@
+from port_ocean.utils import http_async_client
+import httpx
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from enum import StrEnum
+import asyncio
+
+from port_ocean.context.event import event
+
+PAGE_SIZE = 100
+
+
+class ObjectKind(StrEnum):
+    PROJECT = "project"
+    AUDITLOG = "auditlog"
+    FEATURE_FLAG = "flag"
+    ENVIRONMENT = "environment"
+
+
+class LaunchDarklyClient:
+    def __init__(
+        self, api_token: str, launchdarkly_url: str = "https://app.launchdarkly.com"
+    ):
+        self.api_url = f"{launchdarkly_url}/api/v2"
+        self.api_token = api_token
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_token}",
+            "Content-Type": "application/json",
+        }
+
+    async def get_paginated_resource(
+        self, kind: str, resource_path: str | None = None, limit: int = PAGE_SIZE
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        kind = kind + "s"
+        url = kind if not resource_path else f"{kind}/{resource_path}"
+        url = url.replace("auditlogs", "auditlog")

```

Comment:
how the `auditlogs` is relevant for all resources using this method?

Reply:
url = url.replace("auditlogs", ObjectKind.AUDITLOG) if kind == ObjectKind else url